File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hmap = dict()

        for num in nums:
            if num not in hmap:
                hmap[num] = 0
            hmap[num] = hmap[num] +1
        
        for num in nums:
            if hmap[num] > len(nums)//2:
                return num

File: Data Structures & Algorithms/maximum-sum-circular-subarray/submission-1.java
class Solution {
    public int maxSubarraySumCircular(int[] nums) {
        int curMax = 0;
        int curMin = 0;
        int minSum = nums[0];
        int maxSum = nums[0];
        int total = 0;
        for (int num : nums) {
            curMax = Math.max(num, curMax + num);
            maxSum = Math.max(curMax, maxSum);
            curMin = Math.min(num, curMin + num);
            minSum = Math.min(curMin, minSum);
            total += num;
        }
        return maxSum > 0 ? Math.max(maxSum, total - minSum) : maxSum;
    }
}